test/mocks/stream_info/mocks.cc

```diff
@@ -184,6 +184,13 @@ MockStreamInfo::MockStreamInfo()
       .WillByDefault(Invoke([this](const BytesMeterSharedPtr& downstream_bytes_meter) {
         downstream_bytes_meter_ = downstream_bytes_meter;
       }));
+  ON_CALL(*this, setDownstreamTransportFailureReason(_))
+      .WillByDefault(Invoke([this](absl::string_view failure_reason) {
+        downstream_transport_failure_reason_ = std::string(failure_reason);
+      }));
+  ON_CALL(*this, downstreamTransportFailureReason()).WillByDefault(Invoke([this]() {
+    return absl::string_view(downstream_transport_failure_reason_);
```

Comment:
Ditto. 
The key here is returning live value of `downstream_transport_failure_reason_`, string_view conversion here is redundant.

Alternatively you can use  `ReturnPointee` to return the live value, but it is up to you.
```
 ON_CALL(*this, downstreamTransportFailureReason())
      .WillByDefault(ReturnPointee(&downstream_transport_failure_reason_));
```

Reply:
Thanks, done.